webapp/lib/app/nook/view.dart

```diff
@@ -936,7 +939,19 @@ class ConversationListPanelView {
     panelFilters.append(conversationTurnsFilter.conversationFilter);
   }
 
-  void updateConversationList(Set<Conversation> conversations) {
+  void onAddConversation(ConversationSummary item) {
+    var currentDateSeparatorString = dateStringForSeparator(item._dateTime);
+    var sortOrder = controller.conversationSortOrder;
+    if (sortOrder == UIConversationSort.alphabeticalById) {
+      item._toggleDateSeparator(false);
+      return;
+    }
```

Comment:
I suggest you create a boolean `shouldShowDateSeparator` on the `ConversationListPanelView`, which the controller sets to true or false when the sort order is changed, and here you test.

Otherwise you have part of the logic specific to the sort order in the view, instead of all in one place in the controller. Whether the conversation list will show the date separator or not then becomes part of the API for the UI element, rather than embedding the logic to compute it.